File: iosp1/P1/P1/Network/SessionManager.swift
import Foundation

class SessionManager {
    static let shared = SessionManager()
    
    private let defaults = UserDefaults.standard
    
    private let keyToken = "auth_token"
    private let keyDeviceId = "device_id"
    private let keyProfileId = "profile_id"
    private let keyUserName = "user_name"
    
    private init() {}
    
    func saveToken(_ token: String) {
        saveTokens(access: token, refresh: nil)
    }

    func saveTokens(access: String, refresh: String?) {
        defaults.set(access, forKey: keyToken)
        defaults.set(access, forKey: "access_token")
        if let refresh { defaults.set(refresh, forKey: "refresh_token") }
    }

    func getRefreshToken() -> String? {
        defaults.string(forKey: "refresh_token")
    }

    func getAccessToken() -> String? {
        defaults.string(forKey: "access_token") ?? defaults.string(forKey: keyToken)
    }

    func getToken() -> String? {
        getAccessToken()
    }

    func clearToken() {
        defaults.removeObject(forKey: keyToken)
        defaults.removeObject(forKey: "access_token")
        defaults.removeObject(forKey: "refresh_token")
    }
    
    func saveProfileId(_ profileId: String) {
        defaults.set(profileId, forKey: keyProfileId)
    }
    
    func getProfileId() -> String? {
        return defaults.string(forKey: keyProfileId)
    }
    
    func saveUserName(_ name: String) {
        defaults.set(name, forKey: keyUserName)
    }
    
    func getUserName() -> String? {
        return defaults.string(forKey: keyUserName)
    }
    
    func getOrCreateDeviceId() -> String {
        if let id = defaults.string(forKey: keyDeviceId) {
            return id
        } else {
            let newId = "device_\(UUID().uuidString.prefix(8))"
            defaults.set(newId, forKey: keyDeviceId)
            return newId
        }
    }
    
    func clearSession() {
        defaults.removeObject(forKey: keyToken)
        defaults.removeObject(forKey: keyProfileId)
        defaults.removeObject(forKey: keyUserName)
    }
}

File: androidp1/app/src/main/java/com/preappointment1/app/data/SessionManager.kt
package com.preappointment1.app.data

import android.content.Context
import android.content.SharedPreferences

object SessionManager {
    private const val PREFS_NAME = "lpm_prefs"
    private const val KEY_TOKEN = "auth_token"
    private const val KEY_ACCESS_TOKEN = "access_token"
    private const val KEY_REFRESH_TOKEN = "refresh_token"
    private const val KEY_DEVICE_ID = "device_id"
    private const val KEY_PROFILE_ID = "profile_id"
    private const val KEY_USER_NAME = "user_name"

    private lateinit var prefs: SharedPreferences

    fun init(context: Context) {
        prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    fun saveToken(token: String) {
        saveTokens(accessToken = token, refreshToken = null)
    }

    fun saveTokens(accessToken: String, refreshToken: String?) {
        prefs.edit()
            .putString(KEY_ACCESS_TOKEN, accessToken)
            .putString(KEY_TOKEN, accessToken)
            .apply()
        if (refreshToken != null) {
            prefs.edit().putString(KEY_REFRESH_TOKEN, refreshToken).apply()
        }
    }

    fun getRefreshToken(): String? {
        if (!this::prefs.isInitialized) return null
        return prefs.getString(KEY_REFRESH_TOKEN, null)
    }

    fun getToken(): String? = getAccessToken()

    fun getAccessToken(): String? {
        if (!this::prefs.isInitialized) return null
        return prefs.getString(KEY_ACCESS_TOKEN, null)
            ?: prefs.getString(KEY_TOKEN, null)
    }

    fun clearToken() {
        if (!this::prefs.isInitialized) return
        prefs.edit()
            .remove(KEY_TOKEN)
            .remove(KEY_ACCESS_TOKEN)
            .remove(KEY_REFRESH_TOKEN)
            .apply()
    }

    fun saveProfileId(profileId: String) {
        prefs.edit().putString(KEY_PROFILE_ID, profileId).apply()
    }

    fun getProfileId(): String? {
        if (!this::prefs.isInitialized) return null
        return prefs.getString(KEY_PROFILE_ID, null)
    }

    fun saveUserName(name: String) {
        prefs.edit().putString(KEY_USER_NAME, name).apply()
    }

    fun getUserName(): String? {
        if (!this::prefs.isInitialized) return null
        return prefs.getString(KEY_USER_NAME, null)
    }

    fun getOrCreateDeviceId(): String {
        var id = prefs.getString(KEY_DEVICE_ID, null)
        if (id == null) {
            id = "device_" + java.util.UUID.randomUUID().toString().take(8)
            prefs.edit().putString(KEY_DEVICE_ID, id).apply()
        }
        return id
    }

    fun clearSession() {
        if (!this::prefs.isInitialized) return
        prefs.edit().clear().apply()
    }
}
